lib/class-wp-rest-menu-locations-controller.php

```diff
@@ -0,0 +1,269 @@
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$registered_menus = get_registered_nav_menus();
+		if ( ! array_key_exists( $request['location'], $registered_menus ) ) {
+			return new WP_Error( 'rest_menu_location_invalid', __( 'Invalid menu location.' ), array( 'status' => 404 ) );
+		}
+
+		$location              = new stdClass();
+		$location->name        = $request['location'];
+		$location->description = $registered_menus[ $location->name ];
+
+		$data = $this->prepare_item_for_response( $location, $request );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Prepares a menu location object for serialization.
+	 *
+	 * @param stdClass        $location Post status data.
+	 * @param WP_REST_Request $request  Full details about the request.
+	 *
+	 * @return WP_REST_Response Post status data.
+	 */
+	public function prepare_item_for_response( $location, $request ) {
+		$locations = get_nav_menu_locations();
+		$menu_id   = ( isset( $locations[ $location->name ] ) ) ? $locations[ $location->name ] : 0;
+		$data      = array(
+			'name'        => $location->name,
+			'description' => $location->description,
+			'menu_id'     => $menu_id,
+		);
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		$response = rest_ensure_response( $data );
+
+		$response->add_links( $this->prepare_links( $location ) );
+
+		/**
+		 * Filters a menu location returned from the REST API.
+		 *
+		 * Allows modification of the menu location data right before it is
+		 * returned.
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param object           $location The original status object.
+		 * @param WP_REST_Request  $request  Request used to generate the response.
+		 */
+		return apply_filters( 'rest_prepare_menu_location', $response, $location, $request );
+	}
+
+	/**
+	 * Retrieves the menu location's schema, conforming to JSON Schema.
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'menu-location',
+			'type'       => 'object',
+			'properties' => array(
+				'name'        => array(
+					'description' => __( 'The name of the menu location.' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'description' => array(
+					'description' => __( 'The description of the menu location.' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'menu_id'     => array(
```

Comment:
I'd suggest changing this to just `menu` or `assigned_menu`. The REST API generally doesn't add an `_id` suffix.

Reply:
I agree, this should be `menu`